cmd/collector/main.go

```diff
@@ -130,6 +139,7 @@ func main() {
 	config.AddFlags(
 		v,
 		command,
+		flags.AddConfFileFlag,
 		flags.AddFlags,
```

Comment:
any reason not to have config flag added as part of AddFlags? It would probably also avoid duplicating the code in L65-71 in all binaries, only do it in `sFlags := new(flags.SharedFlags).InitFromViper(v)`

Reply:
I did a split for two reasons:

1. agent does use `flags.AddFlags`
2. `flags.AddFlags` can be specified in config file, so there would be twice `sFlags := new(flags.SharedFlags).InitFromViper(v)`

Reply:
I see - makes sense. In that case can we move L65-71 into that package too, so that we can have just a one liner in the mains?  E.g. something like `flags.TryProcessExternalConfig(v, logger)`

Reply:
also, by doing that we can make `flags.ExternalConfFlags` type private. I suggest renaming it to `ConfigFile(Flag)` ("external" is non-informative)